src/ui/component/fileList/view.jsx

```diff
@@ -1,165 +1,66 @@
-        fileInfos.slice().sort(({ file_name: fileName1 }, { file_name: fileName2 }) => {
-          const fileName1Lower = fileName1.toLowerCase();
-          const fileName2Lower = fileName2.toLowerCase();
-          if (fileName1Lower < fileName2Lower) {
-            return -1;
-          } else if (fileName2Lower > fileName1Lower) {
-            return 1;
-          }
-          return 0;
-        }),
-    };
+  function handleSortChange() {
+    setCurrentSort(currentSort === SORT_NEW ? SORT_OLD : SORT_NEW);
   }
 
-  getChannelSignature = (fileInfo: { pending: boolean } & FileListItem) => {
-    if (fileInfo.pending) {
-      return undefined;
-    }
-
-    return fileInfo.channel_claim_id;
-  };
-
-  handleSortChanged(event: SyntheticInputEvent<*>) {
-    this.props.setFileListSort(this.props.page, event.target.value);
-  }
+  const sortedUris = uris && defaultHeader && currentSort === SORT_OLD ? uris.reverse() : uris;
 
-  sortFunctions: {};
-
-  render() {
-    const { fileInfos, hideFilter, sortBy } = this.props;
-
-    const content = [];
-    if (!fileInfos) {
-      return null;
-    }
-
-    this.sortFunctions[sortBy](fileInfos).forEach(fileInfo => {
-      const { name: claimName, claim_name: claimNameDownloaded, claim_id: claimId, txid, nout, isNew } = fileInfo;
-      const uriParams = {};
-
-      // This is unfortunate
-      // https://github.com/lbryio/lbry/issues/1159
-      const name = claimName || claimNameDownloaded;
-      uriParams.contentName = name;
-      uriParams.claimId = claimId;
-      const uri = buildURI(uriParams);
-      const outpoint = `${txid}:${nout}`;
-
-      // See https://github.com/lbryio/lbry-desktop/issues/1327 for discussion around using outpoint as the key
-      content.push(<FileCard key={outpoint} uri={uri} isNew={isNew} />);
-    });
-
-    return (
-      <section>
-        {!hideFilter && (
-          <Form>
-            <FormField label={__('Sort by')} type="select" value={sortBy} onChange={this.handleSortChanged}>
-              <option value={SORT_OPTIONS.DATE_NEW}>{__('Newest First')}</option>
-              <option value={SORT_OPTIONS.DATE_OLD}>{__('Oldest First')}</option>
-              <option value={SORT_OPTIONS.TITLE}>{__('Title')}</option>
+  return (
+    <section className="file-list">
+      {(header || defaultHeader || sort) && (
+        <div className="file-list__header">
+          {defaultHeader ? (
+            <FormField type="select" name="trending_sort" value={currentSort} onChange={handleSortChange}>
+              <option value={SORT_NEW}>{__('Newest First')}</option>
+              <option value={SORT_OLD}>{__('Oldest First')}</option>
             </FormField>
-          </Form>
-        )}
-
-        <section className="media-group--list">
-          <div className="card__list">{content}</div>
-        </section>
-      </section>
-    );
-  }
+          ) : (
+            header
+          )}
+          <div className="file-list__sort">{sort}</div>
```

Comment:
Not really a fan of `sort`. This is the item on the right side of the header. Currently the two pages that use this are the homepage (trending/best/new) and search feedback (satisfied with results?)

Sort isn't really a good name.

Reply:
`headerAltControls`? `headerSecondaryControls`? or same names without the word header?